infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/LogReplicationAckReader.java

```diff
@@ -442,8 +442,11 @@ public void markSyncStatus(SyncStatus status) {
     /**
      * Start periodic replication status update task (completion percentage)
      */
-    public void startSyncStatusUpdatePeriodicTask() {
+    public void startSyncStatusUpdatePeriodicTask(SyncType type) {
         log.info("Start sync status update periodic task");
+        if (type != null) {
+            setSyncType(type);
+        }
         lastAckedTsPoller = Executors.newSingleThreadScheduledExecutor(
```

Comment:
I know it was not introduced by your PR. I think it doesn't make sense to default the `SyncType lastSyncType = SyncType.LOG_ENTRY` as of now. From what I understand, the semantic of the name "lastSyncType" is before our current round of log replication, last time it should end with log entry sync.

But actually we get last sync type from status table instead of given by this field, and we are now setting it to snapshot sync when exiting from InitializedState (i.e. if not to InLogEntrySyncState directly). So I think it could be default with SNAPSHOT and renaming to something like "ackedSyncType" ...